modules/global/src/com/haulmont/cuba/core/sys/AbstractViewRepository.java

```diff
@@ -438,9 +446,15 @@ protected View deployView(Element rootElem, Element viewElem, Set<ViewInfo> visi
         View v = retrieveView(metaClass, viewName, visited);
         boolean overwrite = Boolean.parseBoolean(viewElem.attributeValue("overwrite"));
 
-        String ancestor = viewElem.attributeValue("extends");
-        if (!overwrite) {
-            overwrite = StringUtils.equals(ancestor, viewName);
+        String extended = viewElem.attributeValue("extends");
+        List<String> ancestors = null;
+
+        if (StringUtils.isNotBlank(extended)) {
+            ancestors =  Arrays.asList(extended.split(","));
+        }
+
+        if (!overwrite && ancestors != null) {
+            overwrite = ancestors.stream().anyMatch(ancestor -> StringUtils.equals(ancestor, viewName));

```

Comment:
Wrap lambda expression to the second line. 
Use Objects.equals instead of StringUtils.equals